### DISCUSSION TITLE
The case for Linux System Services

### DISCUSSION THREAD

**Discussion by v1gnesh** [zopencommunity/meta#325] [General]
One thing that I've come to realize recently is that as many (Go/C/C++) applications are ported, they usually run as processes.
And what's the normal way of handling lifecycle of processes... `app.service` systemd units.

Today, we have USS, and that runs as the OMVS address space (I know there's also the BPX* stuff; do correct my understanding).
zCX is already capable of running Linux inside of it. 
Is it reasonable to think that IBM can bless an instance of zCX running RedHat CoreOS, give it special capabilities & privileges, and call it LSS?
zCX is already "free", and most of it goes on zIIP.
By relying on technologies that were necessary for zCX to happen, rather than work afresh to "upgrade" USS to LSS, is it conceivable that a blessed zCX can become LSS?
There are of course things like dataset (storage) access from within zCX, and DSFS will grow (assuming) to address that.

The advantages of having Linux as a subsystem are:
- Get dual use out of RedHat Package Manager (RPM), an asset that IBM already owns
- Use the same versioning logic of OS package management repositories
Ex: In numactl-devel-2.0.16-2.fc38.s390x.html,
2.0.16 - actual version
2 - 2nd release of this package
fc38 - for this OS version
s390x - for this arch
- Achieve better cohesion with Red Hat software development in general. RH developers can consider LSS as another target

I may be over-simplifying things as I surely don't know everything that's involved.
Please weigh in, does this make sense?

**Comment by AnthonyGiorgio** [comment]
I think the biggest problem with this approach is that zCX is a separate operating system, and is essentially isolated from z/OS.  There would need to be a lot of work done to get it to the point where it is a peer of OMVS, and that might be more work than just adding the required interfaces.

**Reply by v1gnesh** [reply]
> ... and that might be more work than just adding the required interfaces.

Exactly... makes me wonder why zCX is necessary, and not an OMVS-like subsystem capable of running & managing containers.
I do understand that not everyone can get LoZ, or running LoZ itself entails different things in different orgs.
So zCX is a zOS-managed, zOS-native way of running containers or Openshift.

But now, there's a hermetic zCX running in zOS, with limited integration to the base OS, when there's also a need for zOS-native containers... which is going to require re-work.

Of course I don't know all the details, just trying to understand it as an outsider.
All I can think of is... zCX can run linux containers, zOS should soon get zOS-native containers.
Would it have been hard (harder than creating zCX + zOS containers) to uplift a part of zOS itself (USS to LSS), so it maintains tight integration to zOS, and is therefore capable of running linux & zOS containers.
